kernel/src/vm/vmar/mod.rs

```diff
@@ -690,6 +705,114 @@ impl Vmar_ {
         &self.vm_space
     }
 
+    fn read_maybe_remote(&self, vaddr: Vaddr, writer: &mut VmWriter) -> Result<usize> {
```

Comment:
I think `read_remote`/`write_remote` is good enough. Remote operations should be usable locally by default, although it may not very efficient.